server/src/main/java/org/opensearch/identity/noop/NoopSubject.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.noop;
+
+import java.security.Principal;
+import java.util.Objects;
+
+import org.opensearch.identity.Subject;
+import org.opensearch.identity.Principals;
+
+/**
+ * Implementation of subject that is always authenticated
```

Comment:
should it be 'unauthenticated' here?

Reply:
My current thought, when the Noop implementation is use it shouldn't trip any authentication or permissions checks - as if security was disabled.  This might be less useful over time as we expect disambiguation between different principals.